test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -85,51 +131,301 @@ class SslCredentialsTest : public ::testing::Test {
     }
   }
 
+  absl::StatusOr<std::shared_ptr<const AuthContext>> DoRpc(
+      const SslCredentialsOptions& ssl_options, grpc_ssl_session_cache* cache,
+      bool override_ssl_target_name = true) {
+    ChannelArguments channel_args;
+    if (GetParam().use_session_cache) {
+      channel_args.SetPointer(std::string(GRPC_SSL_SESSION_CACHE_ARG), cache);
+    }
+    if (override_ssl_target_name) {
+      channel_args.SetSslTargetNameOverride("foo.test.google.fr");
+    }
+
+    auto creds = grpc::SslCredentials(ssl_options);
+    SetTlsVersion(GetParam().tls_version, creds.get());
+    std::shared_ptr<Channel> channel = grpc::CreateCustomChannel(
+        server_addr_, grpc::SslCredentials(ssl_options), channel_args);
+
+    auto stub = grpc::testing::EchoTestService::NewStub(channel);
+    grpc::testing::EchoRequest request;
+    grpc::testing::EchoResponse response;
+    request.set_message(kMessage);
+    ClientContext context;
+    context.set_deadline(grpc_timeout_seconds_to_deadline(/*time_s=*/10));
+    grpc::Status result = stub->Echo(&context, request, &response);
+    if (!result.ok()) {
+      return absl::Status(static_cast<absl::StatusCode>(result.error_code()),
+                          result.error_message());
+    }
+    EXPECT_EQ(response.message(), kMessage);
+    return context.auth_context();
+  }
+
+  void ExpectResumedSession(std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("true", ToString(properties[0]));
+  }
+
+  void ExpectNonResumedSession(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("false", ToString(properties[0]));
+  }
+
+  void ExpectOk(const absl::Status& status) {
```

Comment:
Do you think this is cleaner than `EXPECT_TRUE(status.ok())`? It feels like we're adding an extra step

Reply:
Good point, this makes a failure more diagnosable